File: app/src/main/java/com/trinity/a20201031_marcregistre_nycschools/viewmodel/HighSchoolViewModel.java
package com.trinity.a20201031_marcregistre_nycschools.viewmodel;

import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.trinity.a20201031_marcregistre_nycschools.api.RetrofitApi;
import com.trinity.a20201031_marcregistre_nycschools.model.NycHighSchool;
import com.trinity.a20201031_marcregistre_nycschools.repository.HighSchoolRepository;
import java.util.List;
import javax.inject.Inject;

public class HighSchoolViewModel extends ViewModel {


    @Inject public HighSchoolViewModel(){}

    @Inject
    public HighSchoolRepository highSchoolRepository;

    public MutableLiveData<Boolean> isLoading = new MutableLiveData<Boolean>();
    public MutableLiveData<Boolean> showErrorDialog = new MutableLiveData<Boolean>();
    public MutableLiveData<Boolean> showLoading = new MutableLiveData<>();
    public MutableLiveData<Boolean> hideLoading = new MutableLiveData<>();
    public MutableLiveData<Boolean> showErrorView  = new MutableLiveData<>();

    public MutableLiveData<List<NycHighSchool>> nycHighSchoolList   = new MutableLiveData<>();

    public LiveData<List<NycHighSchool>> getNycHighSchoolData(LifecycleOwner ctx, RetrofitApi service) {
        highSchoolRepository.highSchoolData(service).observe(ctx, schools -> {
            isLoading.postValue(false);
            if(schools == null) {
                showErrorDialog.postValue(true);
            } else {
                nycHighSchoolList.postValue(schools);
            }
        });
        return nycHighSchoolList;

    }

    public void showLoadingView(final Boolean shown) {
        if(shown) {
            showLoading.postValue(true);
        } else {
            hideLoading.postValue(false);
        }
    }
}

File: app/src/main/java/com/trinity/a20201031_marcregistre_nycschools/viewmodel/HighSchoolViewModel.kt
package com.trinity.a20201031_marcregistre_nycschools.viewmodel

import android.content.Context
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModel
import com.trinity.a20201031_marcregistre_nycschools.api.RetrofitApi
import com.trinity.a20201031_marcregistre_nycschools.model.NycHighSchool
import com.trinity.a20201031_marcregistre_nycschools.repository.HighSchoolRepository
import javax.inject.Inject

class HighSchoolViewModel @Inject constructor(val repository: HighSchoolRepository) : ViewModel() {
    // @JvmField: exposes a Kotlin property as a field in Java
    @JvmField val isLoading = MutableLiveData<Boolean>()
    @JvmField val showErrorDialog = MutableLiveData<Boolean>()

    private var nycHighSchool: MutableLiveData<List<NycHighSchool>> = MutableLiveData()

    fun getNycHighSchoolData(ctx: Context, service: RetrofitApi) : LiveData<List<NycHighSchool>>? {
        repository.highSchoolData(service).observe(ctx as LifecycleOwner, Observer {
            isLoading.postValue(false)
            if(it == null) {
                showErrorDialog.postValue(true)
            }
            else nycHighSchool.postValue(it)
        })
        return nycHighSchool
    }
}